multiple patches + some cleanup around touched patches

mechoriet opened the pull request:
- port asm code generation for event listeners instead of reflection(should improve performance)

- moved place to check if there are listeners to the SimplePluginManager  just before fired

- port Performance & Concurrency Improvements to Permissions no double lookup

- packet limiter fix compile warning

- avoid double io lookup if file exists 

- don't allow pickup delay to go below 0

- Check channel before reading and change isOpen calls to
 IsActive

remove some more places where blockpositions are used often and can just be 3 int's x,y,z

uRyanxD commented on patches/server/0095-Avoid-double-I-O-operation-on-load-player-file.patch:
```diff
@@ -0,0 +1,24 @@
+             {
+                 file = new File( this.playerDir, UUID.nameUUIDFromBytes( ( "OfflinePlayer:" + entityhuman.getName() ).getBytes( "UTF-8" ) ).toString() + ".dat");
+-                if ( file.exists() )
++                if ( normalFile ) // PandaSpigot
```
This is not the correct line to change (it's a little lower), see: https://github.com/Akarin-project/Akarin/blob/ver/1.16.5/patches/server/0007-Avoid-double-I-O-operation-on-load-player-file.patch

uRyanxD commented on patches/server/0095-Avoid-double-I-O-operation-on-load-player-file.patch:
```diff
@@ -0,0 +1,29 @@
+             // Spigot Start
+             boolean usingWrongFile = false;
+-            if ( org.bukkit.Bukkit.getOnlineMode() && !file.exists() ) // PaperSpigot - Check online mode first
++            boolean normalFile = file.exists() && file.isFile(); // PandaSpigot
```
Remove file.isFile() from here

mechoriet replied:
it won't return from that first if so that second check is always ran
and this was also in the original patch

uRyanxD commented on patches/server/0095-Avoid-double-I-O-operation-on-load-player-file.patch:
```diff
@@ -0,0 +1,29 @@
+             boolean usingWrongFile = false;
+-            if ( org.bukkit.Bukkit.getOnlineMode() && !file.exists() ) // PaperSpigot - Check online mode first
++            boolean normalFile = file.exists() && file.isFile(); // PandaSpigot
++            if ( org.bukkit.Bukkit.getOnlineMode() && !normalFile ) // PaperSpigot - Check online mode first // PandaSpigot
```
To avoid an extra call here

uRyanxD commented on patches/server/0095-Avoid-double-I-O-operation-on-load-player-file.patch:
```diff
@@ -0,0 +1,29 @@
+             // Spigot End
+ 
+-            if (file.exists() && file.isFile()) {
++            if ( normalFile ) { // PandaSpigot
```
and re-add it here

uRyanxD commented on patches/server/0096-Don-t-go-below-0-for-pickupDelay-breaks-picking-up-i.patch:
```diff
@@ -0,0 +1,27 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: mechoriet <kevinworm92@gmail.com>
+Date: Wed, 19 Apr 2023 21:30:30 +0200
+Subject: [PATCH] Don't go below 0 for pickupDelay ,breaks picking up items by
```
Fix space in name here

uRyanxD commented on patches/server/0096-Don-t-go-below-0-for-pickupDelay-breaks-picking-up-i.patch:
```diff
@@ -0,0 +1,27 @@
+             // CraftBukkit start - Use wall time for pickup and despawn timers
+             int elapsedTicks = MinecraftServer.currentTick - this.lastTick;
+             if (this.pickupDelay != 32767) this.pickupDelay -= elapsedTicks;
++            this.pickupDelay = Math.max(0, this.pickupDelay); // PaperSpigot - don't go below 0
```
PaperSpigot?

uRyanxD commented on patches/server/0096-Don-t-go-below-0-for-pickupDelay-breaks-picking-up-i.patch:
```diff
@@ -0,0 +1,27 @@
+         // CraftBukkit start - Use wall time for pickup and despawn timers
+         int elapsedTicks = MinecraftServer.currentTick - this.lastTick;
+         if (this.pickupDelay != 32767) this.pickupDelay -= elapsedTicks;
++        this.pickupDelay = Math.max(0, this.pickupDelay); // PaperSpigot - don't go below 0
```
PaperSpigot?

uRyanxD commented on patches/server/0097-Check-channel-before-reading-and-change-isOpen-calls.patch:
```diff
@@ -0,0 +1,48 @@
+ 
+     public void l() {
+-        if (this.channel != null && !this.channel.isOpen()) {
++        if (this.channel != null && !this.channel.isActive()) {
```
missing comment

uRyanxD commented on patches/server/0097-Check-channel-before-reading-and-change-isOpen-calls.patch:
```diff
@@ -0,0 +1,48 @@
+     public boolean isConnected() { return this.g(); } // PandaSpigot - OBFHELPER
+     public boolean g() {
+-        return this.channel != null && this.channel.isOpen();
++        return this.channel != null && this.channel.isActive();
```
missing comment

uRyanxD commented on patches/server/0097-Check-channel-before-reading-and-change-isOpen-calls.patch:
```diff
@@ -0,0 +1,48 @@
+ 
+     protected void a(ChannelHandlerContext channelhandlercontext, Packet packet) throws Exception {
+-        if (this.channel.isOpen()) {
++        if (this.channel.isActive()) {
```
missing comment

mechoriet commented:
also added a better compat check since before it was checking for every packet if protocolsupport was there which showed up quite high in the hotspots
so it is now a once check and save result for future use after this change I've haven't seen no dropped players and way lower cpu usage since it doesn't have to go and find the plugin in a list again and transform the name to lowercase for every check



also removed some unneeded decompile casts that where visable on structuregenerate Long.valueof( a long value) that where like 20% of the whole method that was ran

uRyanxD commented on patches/server/0100-Check-channel-before-reading-and-change-isOpen-calls.patch:
```diff
@@ -0,0 +1,48 @@
+     }
+ 
+     protected void channelRead0(ChannelHandlerContext channelhandlercontext, Packet object) throws Exception { // CraftBukkit - fix decompile error
++        if(g()) { // PandaSpigot - Check if channel is opened before reading packet
```
PandaSpigot start

uRyanxD commented on patches/server/0100-Check-channel-before-reading-and-change-isOpen-calls.patch:
```diff
@@ -0,0 +1,48 @@
+     protected void channelRead0(ChannelHandlerContext channelhandlercontext, Packet object) throws Exception { // CraftBukkit - fix decompile error
++        if(g()) { // PandaSpigot - Check if channel is opened before reading packet
+         this.a(channelhandlercontext, (Packet) object);
++        }
```
PandaSpigot end

uRyanxD commented on patches/server/0101-world-util-methods-optimization.patch:
```diff
@@ -0,0 +1,172 @@
++    public boolean setTypeAndData(BlockPosition blockposition, IBlockData iblockdata, int i) {
++        // CraftBukkit start - tree generation
++        if (this.captureTreeGeneration) {
++            this.setCapturedBlockType(blockposition, iblockdata, i); // Titanium
```
comment

uRyanxD commented on patches/server/0101-world-util-methods-optimization.patch:
```diff
@@ -0,0 +1,172 @@
++
++        return true;
++    }
++    // Titanium end
```
comment
###
There are some more unnecessary comments in this patch, remove them and add the missing panda ones

uRyanxD commented on patches/server/0101-world-util-methods-optimization.patch:
```diff
@@ -0,0 +1,172 @@
+ 
+     public int b(int i, int j) {
+-        if (i >= -30000000 && j >= -30000000 && i < 30000000 && j < 30000000) {
++        if (isInWorldBounds(i,j)) {
```
unnecessary

uRyanxD commented on patches/server/0101-world-util-methods-optimization.patch:
```diff
@@ -0,0 +1,172 @@
+                         IBlockData iblockdata;
+ 
+-                        if (k1 >= -30000000 && k1 < 30000000 && l1 >= -30000000 && l1 < 30000000) {
++                        if (isInWorldBounds(k1, l1)) {
```
unnecessary

uRyanxD commented on patches/server/0101-world-util-methods-optimization.patch:
```diff
@@ -0,0 +1,172 @@
+ 
+         return k >= -short0 && k <= short0 && l >= -short0 && l <= short0 && this.keepSpawnInMemory; // CraftBukkit - Added 'this.keepSpawnInMemory'
+     }
++    private static boolean isInWorldBounds(int x, int z) {
```
unnecessary

uRyanxD commented on patches/server/0102-reduce-new-blockposition-generation.patch:
```diff
@@ -0,0 +1,69 @@
+ 
+                 if (chunkcoordintpair != null) {
+-                    if (this.world.isLoaded(new BlockPosition(chunkcoordintpair.x << 4, 0, chunkcoordintpair.z << 4))) {
++                    if (this.world.isLoaded(chunkcoordintpair.x << 4, 0, chunkcoordintpair.z << 4)) {
```
comment missing

uRyanxD commented on patches/server/0102-reduce-new-blockposition-generation.patch:
```diff
@@ -0,0 +1,69 @@
+                         chunk = this.world.getChunkAt(chunkcoordintpair.x, chunkcoordintpair.z);
+                         if (chunk.isReady()) {
+                             arraylist.add(chunk);
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
```
several comments missing from this patch

uRyanxD commented on patches/server/0103-Send-nearby-packets-from-world-player-list-not-serve.patch:
```diff
@@ -0,0 +1,122 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: mechoriet <kevinworm92@gmail.com>
+Date: Sat, 29 Apr 2023 20:18:25 +0200
+Subject: [PATCH] =?UTF-8?q?Send=20nearby=20packets=20from=20world=20player?=
```
wtf?
###
There are also several unnecessary, missing, or incorrect comments in this patch

mechoriet replied:
solved name no clue why it did that also made the name a bit more clear
